error compiling sddm with QT6

I have tried to compile sddm with the option BUILD_WITH_QT6 and it failed. First there is an error with UTF-8 not found during the build process . I compile in a chroot. my locale are

root@grat-os2:/mnt/hd/var/log/pkgbuild# locale
LANG=fr_FR.utf8
LC_CTYPE="fr_FR.utf8"
LC_NUMERIC="fr_FR.utf8"
LC_TIME="fr_FR.utf8"
LC_COLLATE="fr_FR.utf8"
LC_MONETARY="fr_FR.utf8"
LC_MESSAGES="fr_FR.utf8"
LC_PAPER="fr_FR.utf8"
LC_NAME="fr_FR.utf8"
LC_ADDRESS="fr_FR.utf8"
LC_TELEPHONE="fr_FR.utf8"
LC_MEASUREMENT="fr_FR.utf8"
LC_IDENTIFICATION="fr_FR.utf8"
LC_ALL=

in the chroot
Chroot path: /mnt/hd
******************************
root@grat-os2:/# locale
LANG=
LC_CTYPE="POSIX"
LC_NUMERIC="POSIX"
LC_TIME="POSIX"
LC_COLLATE="POSIX"
LC_MONETARY="POSIX"
LC_MESSAGES="POSIX"
LC_PAPER="POSIX"
LC_NAME="POSIX"
LC_ADDRESS="POSIX"
LC_TELEPHONE="POSIX"
LC_MEASUREMENT="POSIX"
LC_IDENTIFICATION="POSIX"
LC_ALL=

the last error is

[ 40%] Linking CXX executable ConfigurationTest
make[2]: *** [src/helper/CMakeFiles/sddm-helper-start-x11user.dir/build.make:104: src/helper/CMakeFiles/sddm-helper-start-x11user.dir/xorguserhelper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:633: src/helper/CMakeFiles/sddm-helper-start-x11user.dir/all] Error 2
[ 40%] Built target ConfigurationTest
make: *** [Makefile:166: all] Error 2
=======> ERROR: Building '/usr/ports/perso/sddm/sddm1685614636x86_64.cards.tar.xz' failed.

I add the build log file and my pkgfile for sddm

antoine@grat-os2:/mnt/hd/var/log/pkgbuild$ cat sddm.log
cards create: /usr/ports/perso/sddm package(s) starting build Thu Jun  1 10:17:16 2023

=======> PKGMK_INSTALL: no
=======> PKGMK_WORK_DIR: /tmp/work
=======> PKGMK_SOURCE_DIR: /tmp
=======> WARNING: CLEAN IGNORED
=======> PKGMK_IGNORE_REPO: no
=======> PKGMK_UPDATE_REPO:
=======> PKGMK_IGNORE_FOOTPRINT: yes
=======> PKGMK_IGNORE_MD5SUM: yes
=======> PKGMK_COMPRESS_PACKAGE: yes
=======> PKGMK_COMPRESSION_MODE: xz
=======> name: sddm
=======> version: 0.19.0
=======> release: 6
=======> Building starting...
+ prepare
+ git clone https://github.com/sddm/sddm sddm-0.19.0
Cloning into 'sddm-0.19.0'...
+ cd sddm-0.19.0
+ git checkout 5055884ee6ae652474d6eebb940767a811b5d968
Note: switching to '5055884ee6ae652474d6eebb940767a811b5d968'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 5055884 Introduce Qt6 CI coverage
+ build
+ mkdir build
+ cd build
+ cmake ../sddm-0.19.0 -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LIBEXECDIR=/usr/lib/sddm -DDBUS_CONFIG_DIR=/usr/share/dbus-1/system.d -DDBUS_CONFIG_FILENAME=sddm_org.freedesktop.DisplayManager.conf -DBUILD_MAN_PAGES=ON -DBUILD_WITH_QT6=ON -DUID_MAX=60513
-- The C compiler identification is GNU 13.1.0
-- The CXX compiler identification is GNU 13.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Release build
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2")
-- Performing Test PAM_MESSAGE_CONST
-- Performing Test PAM_MESSAGE_CONST - Success
-- Found PAM: /usr/lib/libpam.so;/usr/lib/libdl.a
-- Checking for module 'xau'
--   Found xau, version 1.0.11
-- Checking for module 'xcb'
--   Found xcb, version 1.15
-- Found XCB: /usr/lib/libxcb.so
-- Checking for module 'xcb-xkb'
--   Found xcb-xkb, version 1.15
-- Found XKB: /usr/lib/libxcb-xkb.so
-- Building Qt 6 version
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /usr/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /usr/include
-- Checking for module 'systemd'
--   Found systemd, version 253
-- Checking for module 'libsystemd'
--   Found libsystemd, version 253
-- Autodetected UID_MIN=1000 from /etc/login.defs
-- The following features have been enabled:

 * systemd, systemd support
 * journald, journald support

-- The following OPTIONAL packages have been found:

 * Qt6CoreTools (required version >= 6.4.3)
 * Qt6Core (required version >= 5.15.0)
 * Qt6DBusTools (required version >= 6.4.3)
 * Qt6DBus (required version >= 5.15.0)
 * OpenGL
 * PkgConfig
 * Qt6GuiTools (required version >= 6.4.3)
 * Qt6Gui (required version >= 5.15.0)
 * Qt6QmlTools (required version >= 6.4.3)
 * Qt6Qml (required version >= 5.15.0)
 * WrapVulkanHeaders
 * Qt6Quick (required version >= 5.15.0)
 * Qt6LinguistTools (required version >= 5.15.0)
 * Qt6Test (required version >= 5.15.0)
 * Qt6QuickTest (required version >= 5.15.0)

-- The following REQUIRED packages have been found:

 * ECM (required version >= 1.4.0)
 * PAM
 * XCB
 * XKB (required version >= 0.5.0)
 * Qt6 (required version >= 5.15.0)

-- The following features have been disabled:

 * elogind, elogind support

-- Configuring done (0.9s)
CMake Warning:
  Value of QML_INSTALL_DIR contained a newline; truncating


-- Generating done (0.0s)
CMake Warning:
  Value of QML_INSTALL_DIR contained a newline; truncating


-- Build files have been written to: /tmp/work/src/build
+ make
[  1%] Generating sddm-greeter.1
[  0%] Generating sddm-state.conf.5
[  3%] Generating sddm.conf.5
[  3%] Generating sddm.1
[  3%] Built target man_sddm-state.conf.rst
[  3%] Built target man_sddm-greeter.rst
[  3%] Built target man_sddm.rst
[  4%] Automatic MOC for target sddm
[  4%] Generating ar.qm
[  4%] Built target man_sddm.conf.rst
[  4%] Built target themes-translation
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/ar.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[  5%] Generating bn.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/bn.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
[  6%] Automatic MOC for target sddm-helper
[  7%] Automatic MOC for target sddm-helper-start-wayland
[  8%] Generating ca.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/ca.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
[  8%] Generating cs.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/cs.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[  9%] Generating da.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/da.qm'...
    Generated 26 translation(s) (26 finished and 0 unfinished)
[  9%] Generating de.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/de.qm'...
    Generated 26 translation(s) (26 finished and 0 unfinished)
[ 10%] Generating es.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/es.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 11%] Generating et.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/et.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
[ 11%] Generating eu.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/eu.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
[ 12%] Generating fi.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/fi.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 12%] Generating fr.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/fr.qm'...
    Generated 26 translation(s) (26 finished and 0 unfinished)
[ 13%] Generating gl.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/gl.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
[ 13%] Generating hi_IN.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/hi_IN.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 14%] Generating he.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/he.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
[ 15%] Generating hu.qm
[ 14%] Built target sddm-helper-start-wayland_autogen
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/hu.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 16%] Generating ie.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/ie.qm'...
    Generated 17 translation(s) (17 finished and 0 unfinished)
[ 16%] Generating is.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/is.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
[ 16%] Automatic MOC for target sddm-helper-start-x11user
[ 17%] Generating it.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/it.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 17%] Generating ja.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/ja.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 18%] Generating ka.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/ka.qm'...
    Generated 26 translation(s) (26 finished and 0 unfinished)
[ 19%] Generating kk.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/kk.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 19%] Generating ko.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/ko.qm'...
    Generated 27 translation(s) (27 finished and 0 unfinished)
[ 20%] Generating lt.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/lt.qm'...
    Generated 26 translation(s) (26 finished and 0 unfinished)
[ 20%] Generating lv.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/lv.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 21%] Generating nb.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/nb.qm'...
    Generated 26 translation(s) (26 finished and 0 unfinished)
[ 22%] Generating nl.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/nl.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 22%] Generating nn.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Warning: dropping duplicate messages in '/tmp/work/src/build/data/translations/nn.qm':

* Context: TextConstants
* Source: Enter your username

* Context: TextConstants
* Source: Select your user and enter password

* Context: TextConstants
* Source: Enter your password

Updating '/tmp/work/src/build/data/translations/nn.qm'...
    Generated 23 translation(s) (23 finished and 0 unfinished)
[ 22%] Built target sddm-helper-start-x11user_autogen
[ 23%] Generating oc.qm
[ 24%] Generating pl.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/oc.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/pl.qm'...
    Generated 19 translation(s) (19 finished and 0 unfinished)
[ 24%] Generating pt_BR.qm
[ 25%] Generating pt_PT.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/pt_BR.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this cau[ 25%] Generating ro.qm
ses problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/pt_PT.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/ro.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 26%] Generating ru.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/ru.qm'...
    Generated 26 translation(s) (26 finished and 0 unfinished)
[ 27%] Automatic MOC for target ConfigurationTest
[ 28%] Generating sk.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/sk.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 28%] Generating sr.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/sr.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 29%] Generating sr@ijekavian.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/sr@ijekavian.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 30%] Generating sr@ijekavianlatin.qm
[ 30%] Built target sddm-helper_autogen
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/sr@ijekavianlatin.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 30%] Generating sr@latin.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/sr@latin.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
    Ignored 2 untranslated source text(s)
[ 31%] Automatic MOC for target QMLThemeConfigTest
[ 32%] Generating sv.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/sv.qm'...
    Generated 26 translation(s) (26 finished and 0 unfinished)
[ 32%] Generating tr.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/tr.qm'...
    Generated 15 translation(s) (15 finished and 0 unfinished)
[ 33%] Generating uk.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/uk.qm'...
    Generated 13 translation(s) (13 finished and 0 unfinished)
[ 33%] Built target ConfigurationTest_autogen
[ 34%] Generating zh_CN.qm
[ 34%] Building CXX object src/helper/CMakeFiles/sddm-helper.dir/sddm-helper_autogen/mocs_compilation.cpp.o
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/zh_CN.qm'...
    Generated 27 translation(s) (27 finished and 0 unfinished)
[ 34%] Generating zh_TW.qm
Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
Qt shall use a UTF-8 locale ("") instead. If this causes problems,
reconfigure your locale. See the locale(1) manual for more information.
Updating '/tmp/work/src/build/data/translations/zh_TW.qm'...
    Generated 27 translation(s) (27 finished and 0 unfinished)
[ 34%] Built target components-translation
[ 35%] Building CXX object src/helper/CMakeFiles/sddm-helper-start-wayland.dir/sddm-helper-start-wayland_autogen/mocs_compilation.cpp.o
In file included from /tmp/work/src/build/src/helper/sddm-helper_autogen/FVSNT77ZGO/../../../../../sddm-0.19.0/src/helper/backend/PamBackend.h:24,
                 from /tmp/work/src/build/src/helper/sddm-helper_autogen/FVSNT77ZGO/moc_PamBackend.cpp:10,
                 from /tmp/work/src/build/src/helper/sddm-helper_autogen/mocs_compilation.cpp:9:
/tmp/work/src/build/src/common/Constants.h:27:96: warning: missing terminating " character
   27 | #define IMPORTS_INSTALL_DIR         "Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
      |                                                                                                ^
/tmp/work/src/build/src/common/Constants.h:30:17: warning: missing terminating " character
   30 | /usr/lib/qt6/qml"
      |                 ^
/tmp/work/src/build/src/common/Constants.h:30:17: error: missing terminating " character
[ 35%] Built target sddm_autogen
[ 35%] Building CXX object src/helper/CMakeFiles/sddm-helper-start-x11user.dir/sddm-helper-start-x11user_autogen/mocs_compilation.cpp.o
[ 35%] Built target QMLThemeConfigTest_autogen
[ 35%] Building CXX object test/CMakeFiles/ConfigurationTest.dir/ConfigurationTest_autogen/mocs_compilation.cpp.o
[ 36%] Building CXX object test/CMakeFiles/ConfigurationTest.dir/ConfigurationTest.cpp.o
/tmp/work/src/build/src/common/Constants.h:28:1: error: 'Qt' does not name a type
   28 | Qt shall use a UTF-8 locale ("") instead. If this causes problems,
      | ^~
[ 37%] Building CXX object src/helper/CMakeFiles/sddm-helper-start-x11user.dir/HelperStartX11User.cpp.o
make[2]: *** [src/helper/CMakeFiles/sddm-helper.dir/build.make:76: src/helper/CMakeFiles/sddm-helper.dir/sddm-helper_autogen/mocs_compilation.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:581: src/helper/CMakeFiles/sddm-helper.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 37%] Building CXX object test/CMakeFiles/ConfigurationTest.dir/__/src/common/ConfigReader.cpp.o
[ 38%] Building CXX object src/helper/CMakeFiles/sddm-helper-start-wayland.dir/HelperStartWayland.cpp.o
In file included from /tmp/work/src/sddm-0.19.0/src/common/MessageHandler.h:24,
                 from /tmp/work/src/sddm-0.19.0/src/helper/HelperStartX11User.cpp:32:
/tmp/work/src/build/src/common/Constants.h:27:96: warning: missing terminating " character
   27 | #define IMPORTS_INSTALL_DIR         "Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
      |                                                                                                ^
/tmp/work/src/build/src/common/Constants.h:30:17: warning: missing terminating " character
   30 | /usr/lib/qt6/qml"
      |                 ^
/tmp/work/src/build/src/common/Constants.h:30:17: error: missing terminating " character
In file included from /tmp/work/src/sddm-0.19.0/src/common/MessageHandler.h:24,
                 from /tmp/work/src/sddm-0.19.0/src/helper/HelperStartWayland.cpp:31:
/tmp/work/src/build/src/common/Constants.h:27:96: warning: missing terminating " character
   27 | #define IMPORTS_INSTALL_DIR         "Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
      |                                                                                                ^
/tmp/work/src/build/src/common/Constants.h:30:17: warning: missing terminating " character
   30 | /usr/lib/qt6/qml"
      |                 ^
/tmp/work/src/build/src/common/Constants.h:30:17: error: missing terminating " character
/tmp/work/src/build/src/common/Constants.h:28:1: error: 'Qt' does not name a type
   28 | Qt shall use a UTF-8 locale ("") instead. If this causes problems,
      | ^~
/tmp/work/src/build/src/common/Constants.h:28:1: error: 'Qt' does not name a type
   28 | Qt shall use a UTF-8 locale ("") instead. If this causes problems,
      | ^~
make[2]: *** [src/helper/CMakeFiles/sddm-helper-start-wayland.dir/build.make:90: src/helper/CMakeFiles/sddm-helper-start-wayland.dir/HelperStartWayland.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:607: src/helper/CMakeFiles/sddm-helper-start-wayland.dir/all] Error 2
[ 39%] Building CXX object src/helper/CMakeFiles/sddm-helper-start-x11user.dir/xorguserhelper.cpp.o
make[2]: *** [src/helper/CMakeFiles/sddm-helper-start-x11user.dir/build.make:90: src/helper/CMakeFiles/sddm-helper-start-x11user.dir/HelperStartX11User.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /tmp/work/src/sddm-0.19.0/src/common/Configuration.h:30,
                 from /tmp/work/src/sddm-0.19.0/src/helper/xorguserhelper.cpp:24:
/tmp/work/src/build/src/common/Constants.h:27:96: warning: missing terminating " character
   27 | #define IMPORTS_INSTALL_DIR         "Detected system locale encoding (ANSI_X3.4-1968, locale "C") is not UTF-8.
      |                                                                                                ^
/tmp/work/src/build/src/common/Constants.h:30:17: warning: missing terminating " character
   30 | /usr/lib/qt6/qml"
      |                 ^
/tmp/work/src/build/src/common/Constants.h:30:17: error: missing terminating " character
/tmp/work/src/build/src/common/Constants.h:28:1: error: 'Qt' does not name a type
   28 | Qt shall use a UTF-8 locale ("") instead. If this causes problems,
      | ^~
In file included from /tmp/work/src/sddm-0.19.0/src/common/Configuration.h:32:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h: In instantiation of 'void SDDM::ConfigEntry< <template-parameter-1-1> >::setValue(const QString&) [with T = QList<QString>]':
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:160:14:   required from here
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:16: error: no match for 'operator>>' (operand types are 'QTextStream' and 'QList<QString>')
  163 |             in >> m_value;
      |             ~~~^~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdebug.h:9,
                 from /usr/include/qt6/QtCore/qcoreapplication.h:18,
                 from /usr/include/qt6/QtCore/QCoreApplication:1,
                 from /tmp/work/src/sddm-0.19.0/src/helper/xorguserhelper.cpp:20:
/usr/include/qt6/QtCore/qtextstream.h:133:18: note: candidate: 'QTextStream& QTextStream::operator>>(QByteArray&)'
  133 |     QTextStream &operator>>(QByteArray &array);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:133:41: note:   no known conversion for argument 1 from 'QList<QString>' to 'QByteArray&'
  133 |     QTextStream &operator>>(QByteArray &array);
      |                             ~~~~~~~~~~~~^~~~~
/usr/include/qt6/QtCore/qtextstream.h:132:18: note: candidate: 'QTextStream& QTextStream::operator>>(QString&)'
  132 |     QTextStream &operator>>(QString &s);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:132:38: note:   no known conversion for argument 1 from 'QList<QString>' to 'QString&'
  132 |     QTextStream &operator>>(QString &s);
      |                             ~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:131:18: note: candidate: 'QTextStream& QTextStream::operator>>(double&)'
  131 |     QTextStream &operator>>(double &f);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:131:37: note:   no known conversion for argument 1 from 'QList<QString>' to 'double&'
  131 |     QTextStream &operator>>(double &f);
      |                             ~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:130:18: note: candidate: 'QTextStream& QTextStream::operator>>(float&)'
  130 |     QTextStream &operator>>(float &f);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:130:36: note:   no known conversion for argument 1 from 'QList<QString>' to 'float&'
  130 |     QTextStream &operator>>(float &f);
      |                             ~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:129:18: note: candidate: 'QTextStream& QTextStream::operator>>(qulonglong&)'
  129 |     QTextStream &operator>>(qulonglong &i);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:129:41: note:   no known conversion for argument 1 from 'QList<QString>' to 'qulonglong&' {aka 'long long unsigned int&'}
  129 |     QTextStream &operator>>(qulonglong &i);
      |                             ~~~~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:128:18: note: candidate: 'QTextStream& QTextStream::operator>>(qlonglong&)'
  128 |     QTextStream &operator>>(qlonglong &i);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:128:40: note:   no known conversion for argument 1 from 'QList<QString>' to 'qlonglong&' {aka 'long long int&'}
  128 |     QTextStream &operator>>(qlonglong &i);
      |                             ~~~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:127:18: note: candidate: 'QTextStream& QTextStream::operator>>(long unsigned int&)'
  127 |     QTextStream &operator>>(unsigned long &i);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:127:44: note:   no known conversion for argument 1 from 'QList<QString>' to 'long unsigned int&'
  127 |     QTextStream &operator>>(unsigned long &i);
      |                             ~~~~~~~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:126:18: note: candidate: 'QTextStream& QTextStream::operator>>(long int&)'
  126 |     QTextStream &operator>>(signed long &i);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:126:42: note:   no known conversion for argument 1 from 'QList<QString>' to 'long int&'
  126 |     QTextStream &operator>>(signed long &i);
      |                             ~~~~~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:125:18: note: candidate: 'QTextStream& QTextStream::operator>>(unsigned int&)'
  125 |     QTextStream &operator>>(unsigned int &i);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:125:43: note:   no known conversion for argument 1 from 'QList<QString>' to 'unsigned int&'
  125 |     QTextStream &operator>>(unsigned int &i);
      |                             ~~~~~~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:124:18: note: candidate: 'QTextStream& QTextStream::operator>>(int&)'
  124 |     QTextStream &operator>>(signed int &i);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:124:41: note:   no known conversion for argument 1 from 'QList<QString>' to 'int&'
  124 |     QTextStream &operator>>(signed int &i);
      |                             ~~~~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:123:18: note: candidate: 'QTextStream& QTextStream::operator>>(short unsigned int&)'
  123 |     QTextStream &operator>>(unsigned short &i);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:123:45: note:   no known conversion for argument 1 from 'QList<QString>' to 'short unsigned int&'
  123 |     QTextStream &operator>>(unsigned short &i);
      |                             ~~~~~~~~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:122:18: note: candidate: 'QTextStream& QTextStream::operator>>(short int&)'
  122 |     QTextStream &operator>>(signed short &i);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:122:43: note:   no known conversion for argument 1 from 'QList<QString>' to 'short int&'
  122 |     QTextStream &operator>>(signed short &i);
      |                             ~~~~~~~~~~~~~~^
/usr/include/qt6/QtCore/qtextstream.h:120:18: note: candidate: 'QTextStream& QTextStream::operator>>(char16_t&)'
  120 |     QTextStream &operator>>(char16_t &ch)
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:120:39: note:   no known conversion for argument 1 from 'QList<QString>' to 'char16_t&'
  120 |     QTextStream &operator>>(char16_t &ch)
      |                             ~~~~~~~~~~^~
/usr/include/qt6/QtCore/qtextstream.h:119:18: note: candidate: 'QTextStream& QTextStream::operator>>(char&)'
  119 |     QTextStream &operator>>(char &ch);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:119:35: note:   no known conversion for argument 1 from 'QList<QString>' to 'char&'
  119 |     QTextStream &operator>>(char &ch);
      |                             ~~~~~~^~
/usr/include/qt6/QtCore/qtextstream.h:118:18: note: candidate: 'QTextStream& QTextStream::operator>>(QChar&)'
  118 |     QTextStream &operator>>(QChar &ch);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:118:36: note:   no known conversion for argument 1 from 'QList<QString>' to 'QChar&'
  118 |     QTextStream &operator>>(QChar &ch);
      |                             ~~~~~~~^~
/usr/include/qt6/QtCore/qtextstream.h:134:18: note: candidate: 'QTextStream& QTextStream::operator>>(char*)'
  134 |     QTextStream &operator>>(char *c);
      |                  ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:134:35: note:   no known conversion for argument 1 from 'QList<QString>' to 'char*'
  134 |     QTextStream &operator>>(char *c);
      |                             ~~~~~~^
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:66:14: note: candidate: 'QTextStream& operator>>(QTextStream&, bool&)'
   66 | QTextStream &operator>>(QTextStream &str, bool &val);
      |              ^~~~~~~~
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:66:49: note:   no known conversion for argument 2 from 'QList<QString>' to 'bool&'
   66 | QTextStream &operator>>(QTextStream &str, bool &val);
      |                                           ~~~~~~^~~
In file included from /usr/include/qt6/QtCore/qfileinfo.h:11,
                 from /usr/include/qt6/QtCore/qdir.h:9,
                 from /usr/include/qt6/QtCore/QDir:1,
                 from /tmp/work/src/sddm-0.19.0/src/common/Configuration.h:27:
/usr/include/qt6/QtCore/qdatetime.h:536:28: note: candidate: 'QDataStream& operator>>(QDataStream&, QDateTime&)'
  536 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QDateTime &);
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qdatetime.h:521:50: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
  521 |     friend Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QDateTime &);
      |                                                  ^~~~~~~~~~~~~
/usr/include/qt6/QtCore/qdatetime.h:534:28: note: candidate: 'QDataStream& operator>>(QDataStream&, QTime&)'
  534 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QTime &);
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qdatetime.h:246:50: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
  246 |     friend Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QTime &);
      |                                                  ^~~~~~~~~~~~~
/usr/include/qt6/QtCore/qdatetime.h:532:28: note: candidate: 'QDataStream& operator>>(QDataStream&, QDate&)'
  532 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QDate &);
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qdatetime.h:180:50: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
  180 |     friend Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QDate &);
      |                                                  ^~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qcalendar.h:10,
                 from /usr/include/qt6/QtCore/qdatetime.h:11:
/usr/include/qt6/QtCore/qlocale.h:1138:28: note: candidate: 'QDataStream& operator>>(QDataStream&, QLocale&)'
 1138 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QLocale &);
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qlocale.h:1138:39: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
 1138 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QLocale &);
      |                                       ^~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qlocale.h:7:
/usr/include/qt6/QtCore/qvariant.h:566:21: note: candidate: 'QDataStream& operator>>(QDataStream&, QVariant::Type&)'
  566 | inline QDataStream &operator>>(QDataStream &s, QVariant::Type &p)
      |                     ^~~~~~~~
/usr/include/qt6/QtCore/qvariant.h:566:45: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
  566 | inline QDataStream &operator>>(QDataStream &s, QVariant::Type &p)
      |                                ~~~~~~~~~~~~~^
/usr/include/qt6/QtCore/qvariant.h:559:28: note: candidate: 'QDataStream& operator>>(QDataStream&, QVariant&)'
  559 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &s, QVariant &p);
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qvariant.h:559:52: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
  559 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &s, QVariant &p);
      |                                       ~~~~~~~~~~~~~^
In file included from /usr/include/qt6/QtCore/qmetatype.h:14,
                 from /usr/include/qt6/QtCore/qobject.h:18,
                 from /usr/include/qt6/QtCore/qeventloop.h:7,
                 from /usr/include/qt6/QtCore/qcoreapplication.h:11:
/usr/include/qt6/QtCore/qdatastream.h:541:21: note: candidate: 'QDataStream& operator>>(QDataStream&, QKeyCombination&)'
  541 | inline QDataStream &operator>>(QDataStream &s, QKeyCombination &combination)
      |                     ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:541:45: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
  541 | inline QDataStream &operator>>(QDataStream &s, QKeyCombination &combination)
      |                                ~~~~~~~~~~~~~^
In file included from /usr/include/qt6/QtCore/qcoreapplication.h:8:
/usr/include/qt6/QtCore/qstring.h:1514:28: note: candidate: 'QDataStream& operator>>(QDataStream&, QString&)'
 1514 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QString &);
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qstring.h:1514:39: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
 1514 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QString &);
      |                                       ^~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qstringview.h:8,
                 from /usr/include/qt6/QtCore/qchar.h:651,
                 from /usr/include/qt6/QtCore/qstring.h:14:
/usr/include/qt6/QtCore/qbytearray.h:580:28: note: candidate: 'QDataStream& operator>>(QDataStream&, QByteArray&)'
  580 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QByteArray &);
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qbytearray.h:580:39: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
  580 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QByteArray &);
      |                                       ^~~~~~~~~~~~~
/usr/include/qt6/QtCore/qchar.h:618:28: note: candidate: 'QDataStream& operator>>(QDataStream&, QChar&)'
  618 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QChar &);
      |                            ^~~~~~~~
  GNU nano 7.2                                                                                                                                                sddm.log
      |                            ^~~~~~~~
/usr/include/qt6/QtCore/qchar.h:618:39: note:   no known conversion for argument 1 from 'QTextStream' to 'QDataStream&'
  618 | Q_CORE_EXPORT QDataStream &operator>>(QDataStream &, QChar &);
      |                                       ^~~~~~~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:388:21: note: candidate: 'template<class Enum> QDataStream& operator>>(QDataStream&, QFlags<T>&)'
  388 | inline QDataStream &operator>>(QDataStream &s, QFlags<Enum> &e)
      |                     ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:388:21: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:16: note:   'QList<QString>' is not derived from 'QFlags<T>'
  163 |             in >> m_value;
      |             ~~~^~~~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:403:1: note: candidate: 'template<class T> std::enable_if_t<std::is_enum<_Tp>::value, QDataStream&> operator>>(QDataStream&, T&)'
  403 | operator>>(QDataStream &s, T &t)
      | ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:403:1: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:13: note:   cannot convert 'in' (type 'QTextStream') to type 'QDataStream&'
  163 |             in >> m_value;
      |             ^~
/usr/include/qt6/QtCore/qdatastream.h:409:63: note: candidate: 'template<class T> QDataStreamIfHasIStreamOperatorsContainer<QList<T>, T> operator>>(QDataStream&, QList<T>&)'
  409 | inline QDataStreamIfHasIStreamOperatorsContainer<QList<T>, T> operator>>(QDataStream &s, QList<T> &v)
      |                                                               ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:409:63: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:13: note:   cannot convert 'in' (type 'QTextStream') to type 'QDataStream&'
  163 |             in >> m_value;
      |             ^~
/usr/include/qt6/QtCore/qdatastream.h:421:62: note: candidate: 'template<class T> QDataStreamIfHasIStreamOperatorsContainer<QSet<T>, T> operator>>(QDataStream&, QSet<T>&)'
  421 | inline QDataStreamIfHasIStreamOperatorsContainer<QSet<T>, T> operator>>(QDataStream &s, QSet<T> &set)
      |                                                              ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:421:62: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:16: note:   'QList<QString>' is not derived from 'QSet<T>'
  163 |             in >> m_value;
      |             ~~~^~~~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:433:73: note: candidate: 'template<class Key, class T> QDataStreamIfHasIStreamOperatorsContainer<QHash<Key, T>, Key, T> operator>>(QDataStream&, QHash<Key, T>&)'
  433 | inline QDataStreamIfHasIStreamOperatorsContainer<QHash<Key, T>, Key, T> operator>>(QDataStream &s, QHash<Key, T> &hash)
      |                                                                         ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:433:73: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:16: note:   'QList<QString>' is not derived from 'QHash<Key, T>'
  163 |             in >> m_value;
      |             ~~~^~~~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:446:78: note: candidate: 'template<class Key, class T> QDataStreamIfHasIStreamOperatorsContainer<QMultiHash<Key, T>, Key, T> operator>>(QDataStream&, QMultiHash<Key, T>&)'
  446 | inline QDataStreamIfHasIStreamOperatorsContainer<QMultiHash<Key, T>, Key, T> operator>>(QDataStream &s, QMultiHash<Key, T> &hash)
      |                                                                              ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:446:78: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:16: note:   'QList<QString>' is not derived from 'QMultiHash<Key, T>'
  163 |             in >> m_value;
      |             ~~~^~~~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:458:72: note: candidate: 'template<class Key, class T> QDataStreamIfHasIStreamOperatorsContainer<QMap<Key, T>, Key, T> operator>>(QDataStream&, QMap<Key, T>&)'
  458 | inline QDataStreamIfHasIStreamOperatorsContainer<QMap<Key, T>, Key, T> operator>>(QDataStream &s, QMap<Key, T> &map)
      |                                                                        ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:458:72: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:16: note:   'QList<QString>' is not derived from 'QMap<Key, T>'
  163 |             in >> m_value;
      |             ~~~^~~~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:470:77: note: candidate: 'template<class Key, class T> QDataStreamIfHasIStreamOperatorsContainer<QMultiMap<Key, T>, Key, T> operator>>(QDataStream&, QMultiMap<Key, T>&)'
  470 | inline QDataStreamIfHasIStreamOperatorsContainer<QMultiMap<Key, T>, Key, T> operator>>(QDataStream &s, QMultiMap<Key, T> &map)
      |                                                                             ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:470:77: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:16: note:   'QList<QString>' is not derived from 'QMultiMap<Key, T>'
  163 |             in >> m_value;
      |             ~~~^~~~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:482:49: note: candidate: 'template<class T1, class T2> QDataStreamIfHasIStreamOperators<T1, T2> operator>>(QDataStream&, std::pair<_T1, _T2>&)'
  482 | inline QDataStreamIfHasIStreamOperators<T1, T2> operator>>(QDataStream& s, std::pair<T1, T2> &p)
      |                                                 ^~~~~~~~
/usr/include/qt6/QtCore/qdatastream.h:482:49: note:   template argument deduction/substitution failed:
/tmp/work/src/sddm-0.19.0/src/common/ConfigReader.h:163:16: note:   'QList<QString>' is not derived from 'std::pair<_T1, _T2>'
  163 |             in >> m_value;
      |             ~~~^~~~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:189:21: note: candidate: 'QTextStream& operator>>(QTextStream&, QTextStreamFunction)'
  189 | inline QTextStream &operator>>(QTextStream &s, QTextStreamFunction f)
      |                     ^~~~~~~~
/usr/include/qt6/QtCore/qtextstream.h:189:68: note:   no known conversion for argument 2 from 'QList<QString>' to 'QTextStreamFunction' {aka 'QTextStream& (*)(QTextStream&)'}
  189 | inline QTextStream &operator>>(QTextStream &s, QTextStreamFunction f)
      |                                                ~~~~~~~~~~~~~~~~~~~~^
[ 40%] Linking CXX executable ConfigurationTest
make[2]: *** [src/helper/CMakeFiles/sddm-helper-start-x11user.dir/build.make:104: src/helper/CMakeFiles/sddm-helper-start-x11user.dir/xorguserhelper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:633: src/helper/CMakeFiles/sddm-helper-start-x11user.dir/all] Error 2
[ 40%] Built target ConfigurationTest
make: *** [Makefile:166: all] Error 2
=======> ERROR: Building '/usr/ports/perso/sddm/sddm1685614636x86_64.cards.tar.xz' failed.


the PKGFILE for sddm

description="QML based X11 and Wayland display manager"
url="https://github.com/sddm/sddm"

packager="Grat-OS Team"
maintainer="Dania"

makedepends=(extra-cmake-modules python-docutils dejavu-ttf qt6-declarative qt6-tools xorg-xauth xorg-server xorg-xinit)

run=(dejavu-ttf xorg-server xorg-xinit)

name=sddm
version=0.19.0
release=6

#Last commit on sddm repo
_commit=5055884ee6ae652474d6eebb940767a811b5d968

PKGMK_KEEP_SOURCES="no"

prepare() {
git clone https://github.com/sddm/sddm $name-$version

cd $name-$version
git checkout ${_commit}
}

build() {
  mkdir build && cd build

  cmake ../$name-$version \
        -DCMAKE_INSTALL_PREFIX=/usr \
        -DCMAKE_BUILD_TYPE=Release  \
        -DCMAKE_INSTALL_LIBEXECDIR=/usr/lib/sddm \
        -DDBUS_CONFIG_DIR=/usr/share/dbus-1/system.d \
        -DDBUS_CONFIG_FILENAME=sddm_org.freedesktop.DisplayManager.conf \
        -DBUILD_MAN_PAGES=ON \
        -DBUILD_WITH_QT6=ON \
        -DUID_MAX=60513

  make
  make DESTDIR=$PKG install

  install -d $PKG/usr/lib/sddm/sddm.conf.d

  $PKG/usr/bin/sddm --example-config > $PKG/usr/lib/sddm/sddm.conf.d/default.conf
# Don't set PATH in sddm.conf
  sed -r 's|DefaultPath=.*|DefaultPath=/usr/local/sbin:/usr/local/bin:/usr/bin|g' -i $PKG/usr/lib/sddm/sddm.conf.d/default.conf
# Unset InputMethod https://github.com/sddm/sddm/issues/952
  sed -e "/^InputMethod/s/qtvirtualkeyboard//" -i $PKG/usr/lib/sddm/sddm.conf.d/default.conf

#Missing directory
  install -vdm755 $PKG/var/lib/sddm

#Symbolic link on pam
mkdir -pv $PKG/etc/pam.d
ln -svf login $PKG/etc/pam.d/system-login
}

the minimum version start with 5.15.0 for qt6 

Best regards

Daniel

I have tried to compile with the patch include in the compile file and I got this error
[ 41%] Generating sessionadaptor.moc
[ 41%] Generating Login1Manager.moc
[ 41%] Generating Login1Seat.moc
[ 42%] Generating Login1Session.moc
[ 42%] Generating displaymanageradaptor.moc
[ 43%] Generating locale1_interface.moc
[ 44%] Generating seatadaptor.moc
[ 45%] Building CXX object src/daemon/CMakeFiles/sddm.dir/sddm_autogen/mocs_compilation.cpp.o
[ 46%] Building CXX object src/helper/CMakeFiles/sddm-helper-start-x11user.dir/HelperStartX11User.cpp.o
In file included from /tmp/work/src/sddm-0.19.0/src/common/MessageHandler.h:24,
                 from /tmp/work/src/sddm-0.19.0/src/helper/HelperStartX11User.cpp:32:
/tmp/work/src/build/src/common/Constants.h:27:37: warning: missing terminating " character
   27 | #define IMPORTS_INSTALL_DIR         "/usr/lib/qt6/qml
      |                                     ^
/tmp/work/src/build/src/common/Constants.h:28:1: warning: missing terminating " character
   28 | "
      | ^
/tmp/work/src/build/src/common/Constants.h:28:1: error: missing terminating " character
[ 47%] Building CXX object src/helper/CMakeFiles/sddm-helper-start-wayland.dir/HelperStartWayland.cpp.o
In file included from /tmp/work/src/sddm-0.19.0/src/common/MessageHandler.h:24,
                 from /tmp/work/src/sddm-0.19.0/src/helper/HelperStartWayland.cpp:31:
/tmp/work/src/build/src/common/Constants.h:27:37: warning: missing terminating " character
   27 | #define IMPORTS_INSTALL_DIR         "/usr/lib/qt6/qml
      |                                     ^
/tmp/work/src/build/src/common/Constants.h:28:1: warning: missing terminating " character
   28 | "
      | ^
/tmp/work/src/build/src/common/Constants.h:28:1: error: missing terminating " character
make[2]: *** [src/helper/CMakeFiles/sddm-helper-start-x11user.dir/build.make:90: src/helper/CMakeFiles/sddm-helper-start-x11user.dir/HelperStartX11User.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:633: src/helper/CMakeFiles/sddm-helper-start-x11user.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 47%] Building CXX object src/helper/CMakeFiles/sddm-helper-start-wayland.dir/waylandsocketwatcher.cpp.o
make[2]: *** [src/helper/CMakeFiles/sddm-helper-start-wayland.dir/build.make:90: src/helper/CMakeFiles/sddm-helper-start-wayland.dir/HelperStartWayland.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 48%] Building CXX object src/daemon/CMakeFiles/sddm.dir/__/common/Configuration.cpp.o
In file included from /tmp/work/src/sddm-0.19.0/src/common/Configuration.h:30,
                 from /tmp/work/src/sddm-0.19.0/src/common/Configuration.cpp:21:
/tmp/work/src/build/src/common/Constants.h:27:37: warning: missing terminating " character
   27 | #define IMPORTS_INSTALL_DIR         "/usr/lib/qt6/qml
      |                                     ^
/tmp/work/src/build/src/common/Constants.h:28:1: warning: missing terminating " character
   28 | "
      | ^
/tmp/work/src/build/src/common/Constants.h:28:1: error: missing terminating " character
[ 48%] Building CXX object src/daemon/CMakeFiles/sddm.dir/__/common/SafeDataStream.cpp.o
[ 48%] Building CXX object test/CMakeFiles/ConfigurationTest.dir/__/src/common/ConfigReader.cpp.o
make[2]: *** [src/daemon/CMakeFiles/sddm.dir/build.make:174: src/daemon/CMakeFiles/sddm.dir/__/common/Configuration.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:607: src/helper/CMakeFiles/sddm-helper-start-wayland.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:474: src/daemon/CMakeFiles/sddm.dir/all] Error 2
[ 49%] Linking CXX executable ConfigurationTest
[ 49%] Built target ConfigurationTest
make: *** [Makefile:166: all] Error 2
=======> ERROR: Building '/usr/ports/perso/sddm/sddm1685694497x86_64.cards.tar.xz' failed.
cards 2.6.4 create: could not parse Pkgfile: While running 'build()'

As we have no directory /usr/lib64 I have included in the cmake option the line -DQT_IMPORTS_DIR=/usr/lib/qt6/qml/  in the pkgfile . The build is OK with both the line and the patch and I have installed sddm and it work's.

During the compile there is just an error 
[ 83%] Building CXX object src/greeter/CMakeFiles/sddm-greeter.dir/GreeterProxy.cpp.o
[ 83%] Building CXX object src/greeter/CMakeFiles/sddm-greeter.dir/KeyboardLayout.cpp.o
/tmp/work/src/sddm-0.19.0/src/greeter/GreeterApp.cpp: In function 'int main(int, char**)':
/tmp/work/src/sddm-0.19.0/src/greeter/GreeterApp.cpp:338:43: warning: 'Qt::AA_EnableHighDpiScaling' is deprecated: High-DPI scaling is always enabled. This attribute no longer has any effect. [-Wdeprecated-declarations]
  338 |         QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qobjectdefs.h:12,
                 from /usr/include/qt6/QtCore/qobject.h:10,
                 from /usr/include/qt6/QtCore/QObject:1,
                 from /tmp/work/src/sddm-0.19.0/src/greeter/GreeterApp.h:24,
                 from /tmp/work/src/sddm-0.19.0/src/greeter/GreeterApp.cpp:21:
/usr/include/qt6/QtCore/qnamespace.h:439:9: note: declared here
  439 |         AA_EnableHighDpiScaling Q_DECL_ENUMERATOR_DEPRECATED_X(
      |         ^~~~~~~~~~~~~~~~~~~~~~~

the minimum version is always 5.15.0 for qt6

I have tested sddm in a VM it works with the X11 session, I will check if the wayland-sessions worked to see if the issue 1707 has gone away

Thank you and best regards

Daniel

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,7 +99,7 @@ find_package(Qt${QT_MAJOR_VERSION} 5.15.0 CONFIG REQUIRED Core DBus Gui Qml Quic
 # find qt5 imports dir
 get_target_property(QMAKE_EXECUTABLE Qt${QT_MAJOR_VERSION}::qmake LOCATION)
 if(NOT QT_IMPORTS_DIR)
-    exec_program(${QMAKE_EXECUTABLE} ARGS "-query QT_INSTALL_QML" RETURN_VALUE return_code OUTPUT_VARIABLE QT_IMPORTS_DIR)
+    execute_process(COMMAND ${QMAKE_EXECUTABLE} -query QT_INSTALL_QML OUTPUT_VARIABLE QT_IMPORTS_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
 
 # Uninstall target

